diff --git a/frege-interpreter-core/src/main/frege/frege/scriptengine/FregeScriptEngine.fr b/frege-interpreter-core/src/main/frege/frege/scriptengine/FregeScriptEngine.fr
index d1f9821..b8f3104 100644
--- a/frege-interpreter-core/src/main/frege/frege/scriptengine/FregeScriptEngine.fr
+++ b/frege-interpreter-core/src/main/frege/frege/scriptengine/FregeScriptEngine.fr
@@ -368,7 +368,7 @@ compile !engine !script !context = do
   (res, newState) <- Interpreter.run (interpret script) config state
   createFregeCompiledScript script engine config res newState
 
-compileReader :: MutableIO ScriptEngine -> Reader -> MutableIO ScriptContext -> IOMutable CompiledScript
+compileReader :: MutableIO ScriptEngine -> MutableIO Reader -> MutableIO ScriptContext -> IOMutable CompiledScript
 compileReader !engine !reader !context = do
   bufReader <- BufferedReader.new reader
   lines <- bufReader.getLines
